src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -377,107 +587,123 @@ private async Task<List<string>> GetRequiredResourceProvidersNotRegisteredAsync(
 
         private async Task ConfigureVmAsync(ConnectionInfo sshConnectionInfo)
         {
-            await ExecuteCommandOnVirtualMachineAsync(sshConnectionInfo, $"sudo mkdir /cromwellazure && sudo chown {configuration.VmUsername} /cromwellazure && sudo chmod ug=rwx,o= /cromwellazure");
-            await CopyInstallationFilesAsync(sshConnectionInfo);
-            await Task.WhenAll(new[] { RunInstallationScriptAsync(sshConnectionInfo), CopyAnyCustomDockerImagesToTheVmAsync(sshConnectionInfo) });
-            await LoadAnyCustomDockerImagesAndCopyDockerComposeFileAsync(sshConnectionInfo);
+            await MountDataDiskOnTheVirtualMachineAsync(sshConnectionInfo);
+            await ExecuteCommandOnVirtualMachineAsync(sshConnectionInfo, $"sudo mkdir -p {CromwellAzureRootDir} && sudo chown {configuration.VmUsername} {CromwellAzureRootDir} && sudo chmod ug=rwx,o= {CromwellAzureRootDir}");
+            await WriteNonPersonalizedFilesToVmAsync(sshConnectionInfo);
+            await RunInstallationScriptAsync(sshConnectionInfo);
+            await HandleCustomImagesAsync(sshConnectionInfo);
+
+            if (!configuration.Update)
+            {
+                await WritePersonalizedFilesToVmAsync(sshConnectionInfo);
+            }
         }
 
-        private async Task CopyInstallationFilesAsync(ConnectionInfo sshConnectionInfo)
+        private async Task<Version> GetInstalledCromwellOnAzureVersionAsync(ConnectionInfo sshConnectionInfo)
         {
-            var startTime = DateTime.UtcNow;
-            var line = RefreshableConsole.WriteLine("Copying installation files to the VM...");
+            var versionString = (await ExecuteCommandOnVirtualMachineAsync(sshConnectionInfo, @"grep -sPo 'CromwellOnAzureVersion=\K(.*)$' /cromwellazure/env-02-internal-images.txt || :")).Output;
```

Comment:
Should this path be /data/cromwellazure/env-02-internal-images.txt ?

Reply:
On 2.0+ system, /cromwellazure is a symlink to /data/cromwellazure, so both work. I'll change it for clarity.